### PR TITLE
More feature on silx.gui.icons

### PR DESCRIPTION
**Pull request by vallsv**
More feature:

- SVG support or any supported Qt format
- SVG as available resource in setup.py
- Cache icons (with weaked structure)
- Fix exception if icon is found

### CODE REVIEW

**Comment by vallsv** [comment]
Wait, there is a typo on my commits

**Comment by vallsv** [comment]
Looks good now

**Comment by vasole** [comment]
Not a big issue, but one should consider using file_ or fileName instead of file as variable name,.

File: silx/gui/test/__init__.py [hunk 1]
```diff
@@ -71,10 +71,11 @@ def suite():
     from ..plot.test import suite as test_plot_suite
     from .test_qt import suite as test_qt_suite
     from .test_console import suite as test_console_suite
+    from .test_icons import suite as test_icons_suite
 
     def suite():
         test_suite = unittest.TestSuite()
         test_suite.addTest(test_qt_suite())
         test_suite.addTest(test_plot_suite())
-        test_suite.addTest(test_console_suite())
```
Comment (PiRK) [inline silx/gui/test/__init__.py]: Accidental line removal?

**Comment by PiRK** [comment]
Don't forget updating MANIFEST.in.

**Comment by vallsv** [comment]
Comment fixes.
There is still things to check on the tests. I remember travis or appveyor was not happy with the raise of the ValueError which is now really raised.

**Comment by vallsv** [comment]
I add `supportedImageFormats` to qt module cause the result of `QImageReader.supportedImageFormats()` is not very convenient and not compatible Python2,3.
Hope tests will pass.

**Comment by vallsv** [comment]
Then i will
- fix the conflict
- and add AnimatedIcon on this pull request
Let me see

**Comment by vallsv** [comment]
That's mergable now

File: examples/simplewidget.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    def createWaitingPushButton2(self):
+        from silx.gui.widgets.WaitingPushButton import WaitingPushButton
+        widget = WaitingPushButton("Push me")
+        widget.setDisabledWhenWaiting(False)
+        widget.clicked.connect(widget.swapWaiting)
+        return widget
+
+
+def main(filenames):
```
Comment (t20100) [inline examples/simplewidget.py]: Why not making it a script and removing def main and if __name__ == "__main__" as this is not meant to be used as a module.

File: examples/simplewidget.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+        main_panel = qt.QWidget(self)
+        main_panel.setLayout(qt.QVBoxLayout())
+        main_panel.layout().addWidget(self.createWaitingPushButton())
+        main_panel.layout().addWidget(self.createWaitingPushButton2())
+
+        self.setCentralWidget(main_panel)
+
+    def createWaitingPushButton(self):
+        from silx.gui.widgets.WaitingPushButton import WaitingPushButton
```
Comment (t20100) [inline examples/simplewidget.py]: I'd put the import at the top of the script

File: examples/simplewidget.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# ###########################################################################*/
+"""Qt Hdf5 widget examples
+.. note:: This module has a dependency on the `h5py <http://www.h5py.org/>`_
+    library, which is not a mandatory dependency for `silx`. You might need
+    to install it if you don't already have it.
+"""
```
Comment (t20100) [inline examples/simplewidget.py]: """This script shows a gallery of simple widgets provided by silx.

It shows the following widgets:
- :class:`WaitingPushButton`: A button with a progress-like waiting animated icon
"""

File: silx/gui/icons.py [hunk 5]
```diff
@@ -29,11 +29,131 @@
...
+_supported_formats = None
+"""Order of file format extension to check"""
+
+
+_process_working = None
+"""Cache an AnimatedIcon for working process"""
+
+
+def get_process_working():
+    """Returns a cached version of the AnimatedIcon pro cess-working.
```
Comment (t20100) [inline silx/gui/icons.py]: API style consistency: get_process_working -> getProcessWorking
The name is confusing to me, I'd call it something like getWaitIcon or getBusyAnimatedIcon

Typo: process-working

File: silx/gui/icons.py [hunk 6]
```diff
@@ -29,11 +29,131 @@
...
+        :param int frameId: Current frame id
+        """
+        if frameId in self.__cacheIcons:
+            self.__currentIcon = self.__cacheIcons[frameId]
+        else:
+            self.__currentIcon = qt.QIcon(self.__movie.currentPixmap())
+            self.__cacheIcons[frameId] = self.__currentIcon
+        self.iconChanged.emit(self.__currentIcon)
+
+    def register(self, object):
```
Comment (t20100) [inline silx/gui/icons.py]: object is a reserved keyword, avoid overloading it

File: silx/gui/widgets/WaitingPushButton.py [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+__license__ = "MIT"
+__date__ = "01/09/2016"
+
+from .. import qt
+from .. import icons
+
+
+class WaitingPushButton(qt.QPushButton):
+    """Button which allow to display a waiting status when, for example,
+    something is still computing.
```
Comment (t20100) [inline silx/gui/widgets/WaitingPushButton.py]: Typo: allows

File: silx/gui/widgets/WaitingPushButton.py [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+        else:
+            qt.QPushButton.__init__(self, parent)
+
+        self.__waiting = False
+        self.__enabled = True
+        self.__icon = icon
+        self.__disabled_when_waiting = True
+
+    def setDisabledWhenWaiting(self, isDisabled):
+        """Enable or disable the auto diable behaviour when the button is waiting.
```
Comment (t20100) [inline silx/gui/widgets/WaitingPushButton.py]: Typo: disable

**Comment by vallsv** [comment]
That's fine. Most of the things are fixed i think

**Pull request merged**